### ISSUE TITLE
Port twisted.words.protocols.jabber to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8423]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#8423|
|Type|enhancement|
|Created|2016-06-05 06:12:53Z|
|Branch|https://github.com/twisted/twisted/tree/py3-jabber-8423|

<details><summary>Searchable metadata</summary>

```
trac-id__8423 8423
type__enhancement enhancement
reporter__ralphm ralphm
priority__normal normal
milestone__Python_3_x Python-3.x
branch__py3_jabber_8423 py3-jabber-8423
branch_author__ralphm ralphm
status__closed closed
resolution__fixed fixed
component__words words
keywords__None None
time__1465107173978093 1465107173978093
changetime__1466979158560762 1466979158560762
version__None None
owner__ralphm ralphm

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm removed owner|
|-|-|

Pull request: https://github.com/twisted/twisted/pull/192.

The various modules were mostly ported one by one, each in their own commit to make reviewing somewhat easier. Builds have run successfully.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm commented|
|-|-|

I merged in the most recent trunk to resolve merge conflicts. Unfortunately, twistedchecker now marks a lot of pylint messages as new: https://buildbot.twistedmatrix.com/builders/twistedchecker/builds/426

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @ralphm|
|-|-|

Changes looks good.

Please not that we now have protected branches... so that for now you will need to create a PR to trigger travis-ci so that the commit status is updated in GitHub

We are working to automate this process... see https://github.com/twisted-infra/braid/issues/208

--------

Only minor comments.

1. twisted/words/protocols/jabber/jid.py JID `__str__`, twisted/words/protocols/jabber/sasl.py sendResponse(data=SOMETHING), twisted/words/protocols/jabber/sasl_mechanisms.py getResponse(challenge=rspauth-directive) lacks coverage and I am not sure that you will be able to merge this

2. In `ca = component.ConnectComponentAuthenticator("cjid", u"secret")` `"cjid"` is native string on purpose?

Please check my comments and merge (if you can :). Thanks!

Feel free to ping me over IRC if you need help with the merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

btw. if you don't have the codecov plugin installed in your preferred browser, you can see the coverage report here https://codecov.io/gh/twisted/twisted/compare/trunk...py3-jabber-8423

... is in the `diff-cov` link

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|GitHub set status to `closed`|
|-|-|

In changeset 8dd2292cac9be85a9f81aef336c35bdb89f46e1c
```
#!CommitTicketReference repository="" revision="8dd2292cac9be85a9f81aef336c35bdb89f46e1c"
Merge py3-jabber-8423: Port twisted.words.protocols.jabber to Python 3

Author: ralphm
Reviewer: adiroiban
Fixes: #8423
```
